relay-server/src/envelope.rs

```diff
@@ -1144,6 +1145,30 @@ impl Envelope {
         self.headers.retention = Some(retention);
     }
 
+    /// Runs transaction parametrization on the DSC trace transaction.
+    ///
+    /// The purpose is for trace rules to match on the parametrized version of the transaction.
+    pub fn parametrize_dsc_transaction(&mut self, rules: &[TransactionNameRule]) {
+        let Some(ErrorBoundary::Ok(dsc)) = &mut self.headers.trace else {
+            return;
+        };
+
+        let Some(transaction) = &mut dsc.transaction else {
+            return;
+        };
+
+        if !transaction.contains('/') {
+            return;
+        }
+
+        let mut annotated = Annotated::new(transaction.clone());
+        normalize_transaction_name(&mut annotated, rules);
+
+        if let Some(parametrized) = annotated.into_value() {
+            *transaction = parametrized;
+        }
```

Comment:
If the normalization decided to erase the transaction name, we should respect that decision, right?

```suggestion
    dsc.transaction = annotated.into_value();
```

Reply:
hm yeah, that makes sense